src/librustc_target/spec/thumbv7a_pc_windows_msvc.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use spec::{LinkerFlavor, Target, TargetOptions, TargetResult, PanicStrategy};
+
+pub fn target() -> TargetResult {
+    let mut base = super::windows_msvc_base::opts();
+
+    base.pre_link_args.get_mut(&LinkerFlavor::Msvc).unwrap().push(
+        "/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\VC\\Tools\\MSVC\\14.11.25503\\lib\\arm".to_string());
```

Comment:
Why this hard-coded path?

Reply:
I was hoping someone might have some suggestions on how to solve this.

The problem is that you must build rust (`x.py build`) in a Visual Studio 2017 developer command prompt *for the host environment* so that it can build code for the host machine. When it tries to build code for the target, it uses the `CC_${target}` and `CFLAGS_${target}` environment variables, which are used to specify a cross compiler and flags for the cross compiler.

However, when it comes time to link, it invokes `link.exe` which ends up invoking the linker for the host machine. I couldn't find anything like `LD_${target}` or `LDFLAGS_${target}` to let me specify the cross linker via environment variables, so the quickest fix was to use `pre_link_args` in the target spec.

Possible solutions include
 - Adding `LD_${target}` and `LDFLAGS_${target}` environment variables
 - Writing code to detect the linker paths based on target architecture
 - Somehow leveraging vcvars.bat

Open to suggestions.

Reply:
>The problem is that you must build rust (x.py build) in a Visual Studio 2017 developer command prompt for the host environment so that it can build code for the host machine. 

Just *don't* run in a VS developer command prompt. Most things should be able to find VS themselves, so if something fails to do that we can just fix that and then cross compiling should work.

Reply:
Thanks for the tip. The following changes allowed me to remove the hardcoded paths:
 - alexcrichton/cc-rs#346
 - alexcrichton/cmake-rs#57